features/flows/elastic/src/main/java/org/opennms/netmgt/flows/elastic/FlowRequestExecutor.java

```
@@ -0,0 +1,89 @@
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.netmgt.flows.elastic;
+
+import java.io.IOException;
+
+import org.opennms.plugins.elasticsearch.rest.RequestExecutorFactory;
+import org.opennms.plugins.elasticsearch.rest.executors.RequestExecutor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.searchbox.action.Action;
+import io.searchbox.client.JestClient;
+import io.searchbox.client.JestResult;
+import io.searchbox.client.config.exception.CouldNotConnectException;
+
+public class FlowRequestExecutor implements RequestExecutor, RequestExecutorFactory {
+
+    private static Logger LOG = LoggerFactory.getLogger(FlowRequestExecutor.class);
+
+    private final long cooldownInMs;
+
+    public FlowRequestExecutor(long cooldownInMs) {
+        if (cooldownInMs < 0) {
+            LOG.warn("Retry cooldown must be >= 0. Using a value of 0 instead.");
+        }
+        this.cooldownInMs = Math.max(0, cooldownInMs);
+    }
+
+    @Override
+    public <T extends JestResult> T execute(JestClient client, Action<T> clientRequest) {
+        try {
+            T result = client.execute(clientRequest);
+            return result;
+        } catch (CouldNotConnectException connectException) {
+            return retry(client, clientRequest);
+        } catch (IOException ex) {
+            return retry(client, clientRequest);
+        } catch (com.google.gson.JsonSyntaxException gsonException) {
+            return retry(client, clientRequest);
+        }
+    }
+
+    public <T extends JestResult> T retry(JestClient client, Action<T> clientRequest) {
+        LOG.debug("Retry request");
+        if (cooldownInMs > 0) {
+            LOG.debug("Sleep " + cooldownInMs + " before retrying");
+            try {
+                Thread.sleep(cooldownInMs);
+            } catch (InterruptedException e) {
```

If we get interrupted while retrying i.e. on application stop, we should probably abort instead of swallowing the exception and continuing to retry.

---

Yeah, makes sense too.